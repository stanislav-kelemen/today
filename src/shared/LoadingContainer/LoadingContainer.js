import React from 'react';
import PropTypes from 'prop-types';

import Preloader from './Preloader/Preloader';
import PreviewPreloader from './Preloader/PreviewPreloader';

const LoadingContainer = ({
  isLoading,
  isEmpty,
  isPreview,
  loadingMessage,
  children
}) => {
  if (isLoading && isPreview) {
    return (
      <div>
        <PreviewPreloader />
      </div>
    );
  }

  if (isLoading) {
    return (
      <div>
        <Preloader />
      </div>
    );
  }
  if (isEmpty) {
    return (
      <div>
        <p>{loadingMessage}</p>
      </div>
    );
  }

  return children;
};

LoadingContainer.propTypes = {
  isLoading: PropTypes.bool.isRequired,
  isEmpty: PropTypes.bool,
  loadingMessage: PropTypes.string,
  children: PropTypes.node.isRequired
};

LoadingContainer.defaultProps = {
  isEmpty: false,
  loadingMessage: ''
};

export default LoadingContainer;
