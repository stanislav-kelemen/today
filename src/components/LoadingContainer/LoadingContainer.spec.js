import React from 'react';
import { shallow } from 'enzyme';

import LoadingContainer from './LoadingContainer';

const mockedProps = {
  isLoading: true,
  isEmpty: false,
  isPreview: false,
  loadingMessage: "No items",
  children: <div />
}

describe('Rendering without crushing', () => {
  let component;
  beforeEach(() => {
    component = shallow(
      <LoadingContainer {...mockedProps}/>
    );
  });

  test('shold render', () => {
    expect(component).toMatchSnapshot();
  });

  test('should render Preloader when isLoading', () => {
    expect(component.find('Preloader')).toHaveLength(1);
  });

  test('should render Preloader when isLoading', () => {
    component.setProps({isEmpty: true, isLoading:false});
    component.update();

    expect(component.text().includes("No items")).toBeTruthy();
  });

});
