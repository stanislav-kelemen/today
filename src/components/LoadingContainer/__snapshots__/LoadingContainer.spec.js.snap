// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering without crushing shold render 1`] = `
<div>
  <Preloader
    overlay={null}
  />
</div>
`;
